ipn/store.go

```diff
@@ -182,7 +184,12 @@ func NewFileStore(path string) (*FileStore, error) {
 		if os.IsNotExist(err) {
 			// Write out an initial file, to verify that we can write
 			// to the path.
-			os.MkdirAll(filepath.Dir(path), 0755) // best effort
+			dirPath := filepath.Dir(path)
+			os.MkdirAll(dirPath, 0755) // best effort
+			if runtime.GOOS == "windows" {
```

Comment:
we shouldn't need this check if you already have a !windows implementation in that paths package.

How about we collapse these 6 lines down to:

```go
    if err := paths.MkStateDir(filepath.Dir(path)); err != nil {
           return nil, fmt.Errorf("creating state directory: %w", err)
    }
```

And remove the `best effort` comment.

Then document that `MkStateDir` both creates it if needed (with the right permissions to start, so no 0755 to 0700 window on Unix too) and fixes its permissions if needed.

Reply:
Ok, will do.